src/view.c

```diff
@@ -1792,6 +1792,58 @@ view_snap_to_region(struct view *view, struct region *region,
 	view_apply_region_geometry(view);
 }
 
+struct output *
+view_find_output(struct view *view, const char *to)
+{
+	assert(view);
+	struct output *new_output;
+	if (!strcasecmp(to, "left")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_LEFT);
+	} else if (!strcasecmp(to, "right")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_RIGHT);
+	} else if (!strcasecmp(to, "up")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_UP);
+	} else if (!strcasecmp(to, "down")) {
+		new_output = view_get_adjacent_output(view, VIEW_EDGE_DOWN);
+	} else {
+		new_output = output_from_name(view->server, to);
+	}
+	return new_output;
+}
+
+void
+view_move_to_output(struct view *view, struct output *output)
+{
+	assert(view);
+	if (view->fullscreen) {
+		return;
+	}
+
+	view_invalidate_last_layout_geometry(view);
+	view_set_output(view, output);
+	if (view_is_floating(view)) {
+		struct wlr_box geo = output_usable_area_in_layout_coords(output);
+		geo.width = MIN(view->pending.width, geo.width);
+		geo.height = MIN(view->pending.height, geo.height);
+		if (rc.placement_policy == LAB_PLACE_AUTOMATIC) {
+			if (placement_find_best(view, &geo)) {
+				view_move_resize(view, geo);
+			} else {
+				view_center(view, NULL);
+			}
+		} else {
+			view_center(view, NULL);
+		}
+	} else if (view->maximized != VIEW_AXIS_NONE) {
+		view_apply_maximized_geometry(view);
```

Comment:
It feels like we might be duplicating code here with `view_apply_special_geometry()` and `view_adjust_for_layout_change()`
Could we refactor any of it.
Don't worry if there's nothing obvious, or if it just over-complicates things.

Reply:
Thx, will take a look tomorrow if I can see some options to de-duplicate.

Reply:
Looking at `view_apply_special_geometry()`, I don't think extracting common things would improve things, there are still subtle differences. `view_adjust_for_layout_change` is also quite different.

Reply:
Okay.